torch/lib/c10d/TCPStore.hpp

```diff
@@ -61,8 +63,24 @@ class TCPStoreDaemon {
 #endif
 };
 
+class ListenThread {
+  public:
+    typedef void (*CallbackFunction)(const std::string&, const std::string&);
+    explicit ListenThread(int listenSocket);
+    ~ListenThread();
+    void addCallback(std::string key, CallbackFunction cb);
+
+  protected:
+    void run();
+    std::thread listenerThread_;
+    int storeListenSocket_;
+    std::unordered_map<std::string, CallbackFunction> keyToCallbacks_;
+    std::vector<int> controlPipeFd_{-1, -1};
+};
+
 class TCPStore : public Store {
  public:
+  typedef void (*CallbackFunction)(const std::string&, const std::string&);
```

Comment:
this is duplicated with the typedef above? Can this be a global typedef in c10d namespace?

Reply:
If this is specific to the watch event, does it make sense to name it `WatchCallback`?